Minio/Helper/utils.cs

```diff
@@ -39,53 +42,54 @@ class utils
-           }
-           if (bucketName[0] == '.' || bucketName[bucketName.Length - 1] == '.')
-           {
-                throw new InvalidBucketNameException(bucketName,"Bucket name cannot start or end with a '.' dot.");
-           }
-           if (bucketName.Any(c => char.IsUpper(c)))
-           {
+            if (bucketName.Trim() == "")
+            {
+                throw new InvalidBucketNameException(bucketName, "Bucket name cannot be empty.");
+            }
+            if (bucketName.Length < 3)
+            {
+                throw new InvalidBucketNameException(bucketName, "Bucket name cannot be smaller than 3 characters.");
+            }
+            if (bucketName.Length > 63)
+            {
+                throw new InvalidBucketNameException(bucketName, "Bucket name cannot be greater than 63 characters.");
+            }
+            if (bucketName[0] == '.' || bucketName[bucketName.Length - 1] == '.')
+            {
+                throw new InvalidBucketNameException(bucketName, "Bucket name cannot start or end with a '.' dot.");
+            }
+            if (bucketName.Any(c => char.IsUpper(c)))
+            {
                 throw new InvalidBucketNameException(bucketName, "Bucket name cannot have upper case characters");
-           }
-           if (invalidDotBucketName.IsMatch(bucketName))
-           {
-                throw new InvalidBucketNameException(bucketName,"Bucket name cannot have successive periods.");
-           }
-           if (!validBucketName.IsMatch(bucketName))
-           {
+            }
+            if (invalidDotBucketName.IsMatch(bucketName))
+            {
+                throw new InvalidBucketNameException(bucketName, "Bucket name cannot have successive periods.");
+            }
+            if (!validBucketName.IsMatch(bucketName))
+            {
                 throw new InvalidBucketNameException(bucketName, "Bucket name contains invalid characters.");
-           }
+            }
         }
         // isValidObjectName - verify object name in accordance with
         //   - http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingMetadata.html
         internal static void validateObjectName(String objectName)
         {
-           if (objectName.Trim() == "")
+            if (objectName.Trim() == "")
             {
                 throw new InvalidObjectNameException(objectName, "Object name cannot be empty.");
             }
-           if (objectName.Length > 1024)
+            if (objectName.Length > 1024)
             {
                 throw new InvalidObjectNameException(objectName, "Object name cannot be greater than 1024 characters.");
             }
-            //c# strings are in utf16 format. they are already in unicode format when they arrive here.
+            // c# strings are in utf16 format. they are already in unicode format when they arrive here.
```

Comment:
The comments here should be brought up to the previous if block and the utf8.ValidString(objectName) lines maybe removed.